tests/buildDeferred.ts

```diff
@@ -0,0 +1,15 @@
+/**
+ * Builds a promise and exposes its `resolve` function such that the promise can
+ * be awaited and resolved outside of the Promise constructor.
+ *
+ * @returns The promise and its `resolve` function.
+ */
+export default function buildDeferred<T>(): {
```

Comment:
This is a neat utility. Might be useful to reuse this elsewhere down the road as well. I've done this often enough in the extension.